packages/mobx-react/__tests__/exp.test.tsx

```diff
@@ -0,0 +1,63 @@
+import React from "react"
+import { observer } from "../src"
+import { render, act } from "@testing-library/react"
+
+/**
+ *  some test suite is too tedious
+ */
+
+afterEach(() => {
+    jest.useRealTimers()
+})
+
+// let consoleWarnMock: jest.SpyInstance | undefined
+// afterEach(() => {
+//     consoleWarnMock?.mockRestore()
+// })
+
+test("TODO", async () => {
+    let renderCount = 0
+    const Child = observer(function Child({ children }) {
+        renderCount++
+        // Accesses Parent's this.props
+        return children()
+    })
+
+    @observer
+    class Parent extends React.Component<any> {
+        // intentionally stable, so test breaks when you disable observable props (comment line 239 in observerClass)
+        renderCallback = () => {
+            return this.props.x
+        }
+        render() {
+            // Access observable props as part of child
+            return <Child>{this.renderCallback}</Child>
+        }
+    }
+
+    function Root() {
+        const [x, setX] = React.useState(0)
+        // Send new props to Parent
+        return (
+            <div onClick={() => setX(x => x + 1)}>
+                <Parent x={x} />
+            </div>
+        )
+    }
+
+    const app = <Root />
+
+    const { unmount, container } = render(app)
+
+    expect(container).toHaveTextContent("0")
+    expect(renderCount).toBe(1)
+
+    await new Promise(resolve => setTimeout(() => resolve(null), 1000))
+    act(() => {
+        console.log("changing state")
+        container.querySelector("div")?.click()
+    })
+    expect(container).toHaveTextContent("1")
+    expect(renderCount).toBe(2)
```

Comment:
It should be 1. The actual number isn't super relevant, it can depend on some React batching implementation details.
The point is that we must not issue an update on ANY component, when a component is already updating.
We use `isUpdating` flag to prevent updating self, but we must avoid calling `forceUpdate` also on other components - in this case on Child component, which is subscribed to `Parent.props`, because we leaked `Parent.props` to `Child` via callback.
It's like calling `childRef.forceUpdate()` during Parent's render. We must not do that. It's an oversight in the original implementation. Once component is updating (Parent recieved new props) the only proper way to propagate change to Child is via props.  This would normally happen if you would create the callback during render.
The test was supposed to demonstrate that while this behavior is not correct, there this unintended "feature" that you can send the same callback ref and still get the update (which is not supposed to happen with `memo`).
So to fix this, we probably just need to make `isUpdating` flag global.

Another way to put it is, that `props`  must be observable for every derivation, except for `observer` derivation - that is any user defined `computed`/`reaction`/`autorun`.

Hope it's a bit clearer now.

Reply:
Maybe a bit less obvious example of usage that will break once it's fixed:
```ts
@observer
class Parent extends Component {
  store = observable({
    get cmp() {
      return this.props.foo + 5;
    }
  })
  render() {
    return <Child store={store} />
  }
}

@observer
class Child extends Component {  
  render() {
    return store.cmp;
  }
}
```

Reply:
@urugator I see what you mean now
yes this is an unintended "feature" that let child display "correct" value, therefore I feel removing it will cause unexpected issue in our fairly large codebase

> that props must be observable for every derivation, except for observer derivation - that is any user defined computed/reaction/autorun.

This is something I can do, as per this [commit](https://github.com/zhantx/mobx/commit/123de8fb005d04d565455a05110ce204eae821d8), I can change `globalState.trackingDerivation` to be an array, and add `isReactObserver`(name TBD) to `IDerivation`

Eventually, during props' get method, I can check if other react observer is tracking it and show a error in console
![image](https://github.com/mobxjs/mobx/assets/14803931/751474f8-62fb-44b2-b831-658fb488d4a6)